File: EMarket/src/main/java/DoAn/B19DCCN445/EMarket/model/Order.java
package DoAn.B19DCCN445.EMarket.model;


import java.util.Collection;
import java.util.Date;
import java.util.Set;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
@Data
@Entity
@Builder
@EntityListeners(AuditingEntityListener.class)
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "orders")
//@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Order {
	@Id
	private String id;
	private Date order_date;
	private Date shipped_date;
	private String note;
	private String ship_address;
	private Double ship_fee;
	private Date paid_date;
	private String order_status;
	@CreatedDate
	@Temporal(TemporalType.TIMESTAMP)
	private Date create_at;
	@LastModifiedDate
	@Temporal(TemporalType.TIMESTAMP)
	private Date update_at;
	
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "customer_id")
	@JsonBackReference("user-orders")
	private User user;
	private String payment_type;
	
//	@JsonManagedReference("order-orderDetails")
	@JsonBackReference("order-orderDetails")
	@OneToMany(mappedBy = "order", cascade = CascadeType.ALL, fetch = FetchType.LAZY )
	private Collection<Order_detail>order_details;
}

File: emarket-ui/src/models/order.js
import Order_detail from "./order_detail";
import Payment_type from "./payment_type";
import User from "./user";

export default class Order {
    constructor(
        id = null,
        note,
        order_date = new Date(),
        shipped_date,
        ship_address,
        city="Ha Noi",
        district="Nam Tu Liem",
        ward="Trung Van",
        houseNo="Dai Linh",
        ship_fee = 10,
        paid_date,
        order_status,
        user = new User(),
        payment_type,
        order_details = [],
        amount=0,
    ) {
        Object.assign(this, {
            id,
            note,
            order_date,
            shipped_date,
            ship_address,
            city,
            district,
            ward,
            houseNo,
            ship_fee,
            paid_date,
            order_status,
            user,
            payment_type,
            order_details,
            amount,
        });
    }
}
